tests/profiles.rs

```diff
@@ -340,3 +340,127 @@ fn timeout() {
+    let leaf_b_srv = server::http1()
+        .route_fn("/load-profile", |_| {
+            Response::builder().status(201).body("".into()).unwrap()
+        })
+        .route_fn("/traffic-split", move |_req| {
+            lb_rsp.fetch_add(1, Ordering::SeqCst);
+            Response::builder()
+                .status(200)
+                .body("leaf-b".into())
+                .unwrap()
+        })
+        .run();
+
+    let ctrl = controller::new_unordered();
+
+    let leaf_a_authority = format!("{}:{}", leaf_a, leaf_a_srv.addr.port());
+    let leaf_b_authority = format!("{}:{}", leaf_b, leaf_b_srv.addr.port());
+
+    let apex_dst_send = ctrl.destination_tx(apex);
+    let leaf_a_dst_send = ctrl.destination_tx(leaf_a_authority.as_str());
+    let leaf_b_dst_send = ctrl.destination_tx(leaf_b_authority.as_str());
+
+    apex_dst_send.send_addr(apex_srv.addr);
+    leaf_a_dst_send.send_addr(leaf_a_srv.addr);
+    leaf_b_dst_send.send_addr(leaf_b_srv.addr);
+
+    let profile_send = ctrl.profile_tx(apex);
+
+    let ctrl = ctrl.run();
+    let proxy = proxy::new().controller(ctrl).run();
+
+    let client = client::http1(proxy.outbound, apex);
+    let apex_metrics = client::http1(proxy.metrics, "localhost");
+
+    let n = 100;
+
+    // 1. Send `n` requests to apex service
+    for _ in 0..n {
+        assert_eq!(client.get("/traffic-split"), "");
+    }
+
+    // 2. Apex service should send `n` responses
+    assert_eq!(apex_responses.load(Ordering::SeqCst), n);
+
+    // 3. Leaf services should send 0 responses
+    assert_eq!(leaf_a_responses.load(Ordering::SeqCst), 0);
+    assert_eq!(leaf_b_responses.load(Ordering::SeqCst), 0);
+
+    // 4. Load service profile that defines traffic split on Apex
+    profile_send.send(controller::profile(
+        vec![
+            controller::route()
+                .request_path("/load-profile")
+                .label("load_profile", "test"),
+            controller::route().request_any(),
+        ],
+        None,
+        vec![
+            controller::traffic_split(leaf_a_authority, 5000),
+            controller::traffic_split(leaf_b_authority, 5000),
+        ],
+    ));
+
+    // 5. Poll metrics until we recognize the profile is loaded
+    loop {
+        assert_eq!(client.get("/load-profile"), "");
+        let m = apex_metrics.get("/metrics");
+        if m.contains("rt_load_profile=\"test\"") {
+            break;
+        }
+
+        ::std::thread::sleep(::std::time::Duration::from_millis(200));
+    }
+    apex_responses.store(0, Ordering::SeqCst);
+
+    // 6. Send `n` requests to apex service
+    for _ in 0..n {
+        let rsp = client.get("/traffic-split");
+        assert!(rsp == "leaf-a" || rsp == "leaf-b");
+    }
+
+    // 7. Apex proxy metrics should assert there are >0 responses
+    assert!(leaf_a_responses.load(Ordering::SeqCst) > 0);
+    assert!(leaf_b_responses.load(Ordering::SeqCst) > 0);
+    assert_eq!(apex_responses.load(Ordering::SeqCst), 0);
```

Comment:
It may be helpful to add one more set of steps to this test that tests changing an existing traffic split. If a new one is applied that sends all traffic to Leaf A, we could assert that `leaf_a_responses == n` and `leaf_b_responses == 0`. Also, if we just remove the traffic split we could instead assert steps 2 and 3.